apps/rush-lib/src/pluginFramework/PhasedScriptActionHooks.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { AsyncSeriesHook, AsyncSeriesWaterfallHook } from 'tapable';
+import { Operation } from '../logic/operations/Operation';
+
+/**
+ * Hooks usable by rush plugins to modify the behavior of a phased script action.
+ * @beta
+ */
+export class PhasedScriptActionHooks {
+  /**
+   * The hook to run after the initial run completes.
+   */
+  public afterRun: AsyncSeriesHook<void> = new AsyncSeriesHook<void>(undefined, 'afterRun');
+  /**
+   * The hook to run after a watch-mode run completes.
+   */
```

Comment:
Watch-mode runs don't "complete." Maybe expand this comment to clarify that this is fired after each set of operations completes after being invoked when a change is detected.

Reply:
Event flow:
1) Initial run starts
2) Initial run completes
3) Rush waits for a file system change
4) Watch run starts
5) Watch run completes
6) Goto (3)